src/Uno.UI/UI/Xaml/DependencyObjectCollection.cs

```diff
@@ -205,7 +212,17 @@ internal List<T>.Enumerator GetEnumeratorFast()
 			=> _list.GetEnumerator();
 
 		private void RaiseVectorChanged(CollectionChange change, int index)
-			=> VectorChanged?.Invoke(this, new VectorChangedEventArgs(change, (uint)index));
+		{
+			if (_vectorChangedHandlers is not null)
```

Comment:
This avoids the event args allocation in case an event subscription happened then it was unsubscribed. In this case, the list will be empty and not null.

```suggestion
			if (_vectorChangedHandlers is { Count: > 0 })
```